src/Glpi/Http/LegacyDropdownListener.php

```diff
@@ -0,0 +1,205 @@
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Http;
+
+use Glpi\Asset\AssetDefinition;
+use Glpi\Asset\AssetModel;
+use Glpi\Asset\AssetType;
+use Glpi\Controller\DropdownController;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+class LegacyDropdownListener implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::REQUEST => ['onKernelRequest', ListenersPriority::LEGACY_LISTENERS_PRIORITIES[self::class]],
+        ];
+    }
+
+    public function onKernelRequest(RequestEvent $event): void
+    {
+        if (!$event->isMainRequest()) {
+            return;
+        }
+
+        $request = $event->getRequest();
+
+        if ($class = $this->findDropdownClass($request)) {
+            $request->attributes->set('_controller', DropdownController::class);
+            $request->attributes->set('class', $class);
+            $request->attributes->set('is_form', \str_ends_with($request->getPathInfo(), '.form.php'));
+        }
+    }
+
+    public function findDropdownClass(Request $request): ?string
+    {
+        $path_info = $request->getPathInfo();
+
+        if ($model_class = $this->findModelClass($path_info)) {
+            return $model_class;
+        }
+
+        if ($asset_class = $this->findAssetclass($request)) {
+            return $asset_class;
+        }
+
+        if ($device_class = $this->findDeviceClass($request)) {
+            return $device_class;
+        }
+
+        return null;
+    }
+
+    private function findModelClass(string $path_info): ?string
+    {
+        $path_regex = '~^/front/(asset/)?(?<basename>.+)(?<form>\.form)?.php~isUu';
```

Comment:
This regex should be adapted to:
 - handle plugins,
 - handle namespaced classes.

```suggestion
        $path_regex = '~^(/(plugins|marketplace)/(?<plugin_key>[a-z0-9]+]))?/front/(?<basename>.+)(?<form>\.form)?.php~isUu';
```

`basename` should be transformed to convert `/` path separators to `\` namespace separators.

Then, there are multiple cases:
1. When there is a namespace separator outside the plugins context, the `basename` should be prefixed by `NS_GLPI` (but this case does not exist, so it can be ignored).
2. When there is a namespace separator inside the plugins context, the `basename` should be prefixed by `NS_PLUG` + `plugin_key` + `basename` (e.g. `/plugins/myplugin/front/mynamespace/foo/bar.php` -> `GlpiPlugin\myplugin\mynamespace\foo\bar`).
3. When there is no namespace separator inside the plugins context, it could still be a namespaced class. There are 2 cases. The class could be either `Plugin` + `plugin_key` + `basename` (e.g. `/plugins/myplugin/front/bar.php` -> `Pluginmypluginbar`), either `NS_PLUG` + `plugin_key` + `basename` (e.g. `/plugins/myplugin/front/bar.php` -> `GlpiPlugin\myplugin\bar`).

Then a call to `DbUtils::fixItemtypeCase()` will permit to get the correct namespace/class case.